components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java

```diff
@@ -30,11 +35,21 @@
  */
 public class ClaimHandlerUtil {
 
-    public static CustomClaimsCallbackHandler getClaimsCallbackHandler(OAuthAppDO oAuthAppDO) {
+    /**
+     * Get the claims callback handler based on the application configuration.
+     *
+     * @param oAuthAppDO OAuth application data object.
+     * @return CustomClaimsCallbackHandler.
+     */
+    public static CustomClaimsCallbackHandler getClaimsCallbackHandler(OAuthAppDO oAuthAppDO)
+            throws IdentityOAuth2Exception {
 
         // If JWT access token OIDC claims separation is enabled and the application is configured to separate OIDC
         // claims, use the JWTAccessTokenOIDCClaimsHandler to handle custom claims.
-        if (isAccessTokenClaimsSeparationFeatureEnabled() && oAuthAppDO.isAccessTokenClaimsSeparationEnabled()) {
+        int appTenantId = IdentityTenantUtil.getLoginTenantId();
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(appTenantId);
+        if (isAccessTokenClaimsSeparationFeatureEnabled() &&
+                isAccessTokenClaimsSeparationEnabledForApp(oAuthAppDO, tenantDomain)) {
```

Comment:
Can   isAccessTokenClaimsSeparationEnabledForApp become true even if isAccessTokenClaimsSeparationFeatureEnabled() false. If so can we please block that?

Reply:
These two are server and app level enabling configs.

Server level -> `isAccessTokenClaimsSeparationFeatureEnabled`
App level -> ` isAccessTokenClaimsSeparationEnabledForApp`